### PR TITLE
tunnelx: init at 2023-07-nix

### PR DESCRIPTION
**Pull request by goatchurchprime**
###### Description of changes

First solo attempt by at getting a cave surveying software package into nixpkgs.  (It's already packaged in Debian.)

It runs directly from the compiled java code (ie directory of .class files), because this is reproducible using a call to `makeWrapper`.  If I batch these up into a .jar file (which is standard practice), then the hash of the result seems to vary.

I am putting it into the `/applications/gis/`directory because it is mapping software.

In the future I hope to move the other two new cave survey packages `/applications/misc/survex` and `/applications/misc/therion` (which I helped @MatthewCroughan to package) into the proper '/applications/gis/`directory.

Homepage for this one: https://github.com/CaveSurveying/tunnelx/

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by MatthewCroughan** [comment]
@goatchurchprime please squash the commit that removes the trailing whitespace into your initial commit so this PR is just a single commit.

File: pkgs/applications/gis/tunnelx/default.nix
Comment (Janik-Haag) [inline pkgs/applications/gis/tunnelx/default.nix:53]:
```suggestion
    description = "A cave surveying software";
```
we don't have the pname at the beginning of a description in nixpkgs

File: pkgs/applications/gis/tunnelx/default.nix
Comment (Janik-Haag) [inline pkgs/applications/gis/tunnelx/default.nix:13]:
```suggestion
stdenv.mkDerivation {
```
you don't need the rec here since you aren't reusing any attribute.

**Review by Janik-Haag — CHANGES_REQUESTED** [review]
Just some small nitpicks

File: maintainers/maintainer-list.nix
Comment (Janik-Haag) [inline maintainers/maintainer-list.nix:5968]: As @MatthewCroughan already said, this should be it's own commit called `maintainers: add goatchurch` otherwise this looks good to me.

**Review by Janik-Haag — APPROVED** [review]
LGTM; just put the maintainer entry into it's own commit.

File: pkgs/applications/gis/tunnelx/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/gis/tunnelx/default.nix:20]:
```suggestion
    rev = "v${version}";
```

File: pkgs/applications/gis/tunnelx/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/gis/tunnelx/default.nix:15]:
```suggestion
  version = "2023-07-nix";
```

File: pkgs/applications/gis/tunnelx/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/gis/tunnelx/default.nix:49]:
```suggestion
      --add-flags "-cp $out/java Tunnel.MainBox" \
      --set SURVEX_EXECUTABLE_DIR ${survex}/bin/ \
      --set TUNNEL_USER_DIR $out/java/
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Also please update the commit message

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 239739` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>tunnelx</li>
  </ul>
</details>

**Review by Janik-Haag — COMMENTED** [review]
Upstream re-released the same version which now has a different commit hash, can you fix that so we can get this merged?

**Review by MatthewCroughan — APPROVED** [review]

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 239739` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tunnelx</li>
  </ul>
</details>

**Comment by goatchurchprime** [comment]
> Upstream re-released the same version which now has a different commit hash, can you fix that so we can get this merged?

Done.  I've no idea how this happened as I'm certain I didn't touch  tunnelx repo after I started on the nixpkgs stuff. Is this an error that one of the bots picks up?

**Comment by Janik-Haag** [comment]
Hm where are you actually getting your release from? It's not a tag in github right?

**Comment by Janik-Haag** [comment]
When starting tunnelx it throws a few warnings
image

**Comment by goatchurchprime** [comment]
> When starting tunnelx it throws a few warnings image

Those startup warnings don't mean anything and have been there for years.  I am going to give this app some love once it gets through this nixos process.

File: pkgs/applications/gis/tunnelx/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/gis/tunnelx/default.nix:45]:
```suggestion
    mkdir -p $out/bin $out/java
    cp -r symbols Tunnel tutorials $out/java
```

**Comment by Janik-Haag** [comment]
@goatchurchprime can you explain where you are pulling the release from? The only tag released on github seems to be from 2019 https://github.com/CaveSurveying/tunnelx/releases/tag/R2019a and your commit hash changed at some point.

**Comment by MatthewCroughan** [comment]
@Janik-Haag since he authors this software, he made a new tag [v2023-07-nix](https://github.com/CaveSurveying/tunnelx/releases/tag/v2023-07-nix) especially for the purpose of this PR

**Comment by Janik-Haag** [comment]
Oh I see, then it's interesting that the commit hash changed at some point without them noticing :/

**Review by drupol — CHANGES_REQUESTED** [review]

**Review by drupol — APPROVED** [review]

**Review by drupol — APPROVED** [review]
LGTM, ready to be merged once the ci is done.

File: pkgs/applications/gis/tunnelx/default.nix
Comment (Artturin) [inline pkgs/applications/gis/tunnelx/default.nix:26]: javac is used in buildPhase so jdk should be in nativeBuildInputs

File: pkgs/applications/gis/tunnelx/default.nix
Comment (Artturin) [inline pkgs/applications/gis/tunnelx/default.nix:33]: missing pre and post runHook's

File: pkgs/applications/gis/tunnelx/default.nix
Comment (Artturin) [inline pkgs/applications/gis/tunnelx/default.nix:37]: missing pre and post runHook's

File: pkgs/applications/gis/tunnelx/default.nix
Comment (Artturin) [inline pkgs/applications/gis/tunnelx/default.nix:31]: runtimeInputs isn't used so this doesn't do anything

File: pkgs/applications/gis/tunnelx/default.nix
Comment (Artturin) [inline pkgs/applications/gis/tunnelx/default.nix:42]: `${lib.getBin survex}`

**Comment by drupol** [comment]
@Artturin I made a new PR fixing the things you reported, please have a look!

**Pull request merged**
